panther_diagnostics/src/system_status.cpp

```diff
@@ -12,33 +12,52 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include "panther_diagnostics/system_status.hpp"
+
 #include <exception>
 #include <filesystem>
 #include <fstream>
 #include <thread>
 
-#include "panther_diagnostics/system_status.hpp"
-
 namespace panther_diagnostics
 {
 
-SystemStatus::SystemStatus() : rclcpp::Node("system_status_node"), diagnostic_updater_(this)
+SystemStatus::SystemStatus(const std::string & node_name)
+: rclcpp::Node(node_name), diagnostic_updater_(this)
 {
-  this->number_of_cpus_ = std::thread::hardware_concurrency();
-  this->cpus_usages_.resize(this->number_of_cpus_ + 1, 0);
-  this->cpus_last_idles_.resize(this->number_of_cpus_ + 1, 0);
-  this->cpus_last_totals_.resize(this->number_of_cpus_ + 1, 0);
+  cpu_cores_ = std::thread::hardware_concurrency();
+  cpu_cores_usages_.resize(cpu_cores_ + 1, 0);
+  cpu_cores_last_idles_.resize(cpu_cores_ + 1, 0);
+  cpu_cores__last_totals_.resize(cpu_cores_ + 1, 0);
 
-  publisher_ = this->create_publisher<panther_msgs::msg::SystemStatus>("system_status", 10);
+  param_listener_ =
+    std::make_shared<system_status::ParamListener>(this->get_node_parameters_interface());
+  params_ = param_listener_->get_params();
+
+  system_status_publisher_ = this->create_publisher<panther_msgs::msg::SystemStatus>(
+    "system_status", 10);
   timer_ = this->create_wall_timer(250ms, std::bind(&SystemStatus::TimerCallback, this));
 
-  diagnostic_updater_.setHardwareID("Overlay RPi");
-  diagnostic_updater_.add("Overlay RPi OS status", this, &SystemStatus::DiagnoseSystem);
+  diagnostic_updater_.setHardwareID("Built-in Computer");
+  diagnostic_updater_.add("OS status", this, &SystemStatus::DiagnoseSystem);
 
   RCLCPP_INFO(this->get_logger(), "Node started");
 }
 
-float SystemStatus::GetTemperature(const std::string & filename) const
+void SystemStatus::TimerCallback()
+{
+  auto message = panther_msgs::msg::SystemStatus();
+  message.header.stamp = this->get_clock()->now();
+  message.header.frame_id = "built_in_computer";
+  message.cpu_percent = GetCPUsUsages(kCPUInfoFilename);
+  message.avg_load_percent = GetCPUMeanUsage();
+  message.cpu_temp = GetCPUTemperature(kTemperatureInfoFilename);
+  message.disc_usage_percent = GetDiskUsage();
+  message.ram_usage_percent = GetMemoryUsage(kMemoryInfoFilename);
+  system_status_publisher_->publish(message);
```

Comment:
I'd appreciate vertical separation improving readability here

```suggestion
  auto message = panther_msgs::msg::SystemStatus();

  message.header.stamp = this->get_clock()->now();
  message.header.frame_id = "built_in_computer";
  message.cpu_percent = GetCPUsUsages(kCPUInfoFilename);
  message.avg_load_percent = GetCPUMeanUsage();
  message.cpu_temp = GetCPUTemperature(kTemperatureInfoFilename);
  message.disc_usage_percent = GetDiskUsage();
  message.ram_usage_percent = GetMemoryUsage(kMemoryInfoFilename);

  system_status_publisher_->publish(message);
```